include/gridtools/stencil_composition/icosahedral_grids/iterate_domain.hpp

```diff
@@ -8,124 +8,105 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #pragma once
-#include <type_traits>
 
-#include "../../common/array.hpp"
 #include "../../common/defs.hpp"
-#include "../../common/gt_assert.hpp"
+#include "../../common/generic_metafunctions/for_each.hpp"
+#include "../../common/host_device.hpp"
 #include "../../common/hymap.hpp"
 #include "../../meta.hpp"
-#include "../caches/cache_metafunctions.hpp"
+#include "../dim.hpp"
 #include "../iterate_domain_aux.hpp"
 #include "../local_domain.hpp"
 #include "../pos3.hpp"
 #include "../sid/multi_shift.hpp"
-#include "dim.hpp"
 
 namespace gridtools {
-    /**
-       This class is basically the iterate domain. It contains the
-       ways to access data and the implementation of iterating on neighbors.
+    /**@brief class managing the memory accesses, indices increment
+
+       This class gets instantiated in the backend-specific code, and has a different implementation for
+       each backend (see CRTP pattern). It is instantiated within the kernel (e.g. in the device code),
```

Comment:
This comment was moved? Please update. (no CRTP?)

Reply:
done